docs/source/posts/2024/2024-07-01-week-5-robin.rst

```diff
@@ -0,0 +1,117 @@
+Week 5: LLM Benchmarking & Architecture Modifications
+=====================================================
+
+.. post:: July 1 2024
+   :author: Robin Roy
+   :tags: google
+   :category: gsoc
+
+Hi, I'm `Robin <https://github.com/robinroy03>`_ and this is my blog about week 5.
+
+This week, we'll take all the things we did in the previous weeks, and quantify them. Benchmarking an LLM is the process of grading the LLM answer. To grade properly, we need good rubrics, so that's what I worked on this week. Also, I made some architectural changes, to make the overall development simple.
+
+Things I did in Week 5
+----------------------
+
+1) **Architectural Update**
+
+Earlier, this was our architecture:
+
+
+   .. raw:: html
+
+      <img src="https://github.com/fury-gl/fury-communication-assets/blob/main/gsoc_2024/7-6-2024-demo-architecture-gsoc-robin-week2.png?raw=true">
+
+This had an obvious issue, all the core logic was inside the Discord Bot. So if I want to say, use the LLM inference for making a GitHub bot, or for benchmarking etc, it wasn't possible. So I decided to cut the LLM logic from Discord Bot and made a new ``LLM Router``. It'll handle all the LLM logic from now on, and we do not directly call any other endpoint other than this one.
+It makes life simple, every input going into the endpoint goes like this:
+
+`{"query": "Render a cube in fury", "llm": "llama3-70b-8192", "knn": "3", "stream": False}`
```

Comment:
this renders like this 
![image](https://github.com/fury-gl/fury/assets/63170874/a06b4e29-9287-43be-a870-b7604ab6bd22)
This doesn't have proper quotation marks for JSON. If you can, replace the curly quotes with straight quotes.